src/cbc_sdk/enterprise_edr/base.py

```diff
@@ -338,14 +487,15 @@ def _validate(self, args):
         if not validated.get("valid"):
             raise ApiError("Invalid query: {}: {}".format(args, validated["invalid_message"]))
 
-    def _search(self, start=0, rows=0):
-        # iterate over total result set, 100 at a time
+    def _search(self, start=None, rows=None):
+        # iterate over total result set, 500 at a time
         args = self._get_query_parameters()
         self._validate(args)
 
-        if start != 0:
+        if start:
             args['start'] = start
-        args['rows'] = self._batch_size
+        if rows:
+            args['rows'] = rows
```

Comment:
Is this suppose to be `if start is None:` and the same for rows?

Reply:
This is intended to be a check for if the params were specified. If they were, we add those keys and vals to our request. If not, we omit them and use the default backend values

Reply:
hmm. Then should the defaults for those not be None? `start=0, rows=20`? Looking through the code it looks like this would fail if not value was set for start or rows since you wouldn't be able to increment or check completion against start and rows

Reply:
Ahh yes that totally makes sense, I hadn't considered that. Good catch!